### DISCUSSION TITLE
sendmail with official Directus Docker image

### DISCUSSION THREAD

**Discussion by sandros94** [directus/directus#18533] [Feature Requests]
So, funny but sad story:
I was experimenting with some Directus stack deployments for collections seeding (you could check a similar version of the docker-compose I'm using [here](https://github.com/directus/directus/discussions/17663)). As usual, as soon as the Directus istance is up and running I change the password right away, even if it is just a dev env (I find it a good practice anyway). But this time I've probably set it with capslock or something, because then I couldn't access at all (wrong password).
I decided to reset the password to only discover that Alpine images do have a `sendmail` but no MTA/SMTP server.

I've found a number of [discussions](https://github.com/directus/directus/discussions/5281) about this topic, and the usual reply is: "use a free smtp provider like sendgrid". But this seems just a bit overkill for just a password reset...
What I would like to understand, and potentially document for a future guide on the Directus docs, is: how could we make the default Docker image of Directus send-only simple system emails like the password recovery?

The aim of this is not about big productions, because in that case I would be surprised if they didn't have an smtp server at hand, but more like for small, indie (potentially no-profit), projects. How should they manage this? Is attaching a full SMTP server the only option?

P.S.: I'm a big fan of the new BSL 1.1 license.

**Comment by rijkvanzanten** [comment]
> how could we make the default Docker image of Directus send-only simple system emails like the password recovery?

The reason I keep recommending using an SMTP server (be it your existing email provider, sendgrid, ses, or whatever) over sendmail is because sendmail is actually very finicky to get up and running to a point where you actually receive emails. To give an example: a lot of users on popular providers (Gmail/Outlook/etc) won't ever receive emails that are coming from an untrusted sendmail execution due to spam and security protections, which are very hard to debug 😬

**Reply by sandros94** [reply]
That was one of my consideration too. 
But wouldn't this be mostly mitigated by proper DNS records such DKIM and DMARC, and maybe TLS?

For the past couple of days this led me in such a deep rabbit hole...

**Reply by rijkvanzanten** [reply]
> But wouldn't this be mostly mitigated by proper DNS records such DKIM and DMARC, and maybe TLS?

Yeah!.. but at that point it is honestly easier to just use Sendgrid or something 😅 

(Also another sendmail quirk I just remembered is that some hosting providers auto-block traffic from sendmail/local smtp servers to prevent spam 🙈 I remember spending days trying to find why sendmail wouldn't work on my digitalocean droplet)

**Reply by sandros94** [reply]
Ok so I've dig around this for some days.

Making the sendmail from the base alpine image is too much customization to be included in the directus project itself (not to mention the plethora of potential user errors).

This said I started investigating into other ready to use tools (open source) like [Mailu](https://github.com/mailu/mailu) and [docker-mailserver](https://github.com/docker-mailserver/docker-mailserver), and discovered them quite easy to setup and use, the _difficult_ part was all the dns shenanigans needed so that popular email provider (aka gmail and co) stop blocking or filtering the emails, but it looks more like a set-once config.

At this point I'm more down to a write an unofficial guide, but I need a bit more experience with the two mentioned (potential) solutions. Currently I would like to focus on other aspects of directus.

For now I'm going to close this just to cleanup the open discussions, I'll update it once I get back on this topic.

**Reply by rijkvanzanten** [reply]
Perfect! Thanks for investigating that 🙂 Sending emails is one of those things in tech that's got surprisingly complex 😅
